src/cares_wrap.cc

```diff
@@ -296,11 +296,6 @@ void node_ares_task::MemoryInfo(MemoryTracker* tracker) const {
 node_ares_task* ares_task_create(ChannelWrap* channel, ares_socket_t sock) {
   auto task = new node_ares_task();
 
-  if (task == nullptr) {
-    /* Out of memory. */
-    return nullptr;
-  }
-
```

Comment:
Could you delete https://github.com/nodejs/node/blob/575e086b6667c0bb8c89153891097938744f1f8a/src/cares_wrap.cc#L338-L343
as well?

Reply:
I can remove them, but they are not quite the same. The `task` is created by method `ares_task_create`. Still can be null as
https://github.com/nodejs/node/blob/c8d3a73c8b673792e315759b70cf4822b64b3e45/src/cares_wrap.cc#L307-L312

Second thought ?

Reply:
Ok, so that will make the second deletion a change in semantics.
IMO that is more appropriate for a separate PR.

So let's keep this PR as it is.